registry/api/v2/descriptors.go

```diff
@@ -637,17 +588,9 @@ var routeDescriptors = []RouteDescriptor{
 									ErrorCodeBlobUnknown,
 								},
 							},
-							{
-								StatusCode:  http.StatusUnauthorized,
-								Description: "The client does not have permission to push to the repository.",
-								Body: BodyDescriptor{
-									ContentType: "application/json; charset=utf-8",
-									Format:      errorsBody,
-								},
-								ErrorCodes: []errcode.ErrorCode{
-									errcode.ErrorCodeUnauthorized,
-								},
-							},
+							unauthorizedResponseDescriptor,
+							repositoryNotFoundResponseDescriptor,
```

Comment:
I was under the impression that the authentication and authorization filters were done before any existence check, in which case a 403 will always be returned if there is no access whether a repository exists or not and 401 is always returned if there is no valid auth method.